tests/test_algorithm.py

```diff
@@ -782,151 +734,96 @@ def test_future_symbol(self):
+             'start_date': cls.asset_starts[1],
+             'end_date': cls.asset_ends[1],
+             'exchange': 'TEST',
+             'asset_name': 'SECOND'},
+            {'symbol': 'BENCH',
+             'start_date': cls.START_DATE,
+             'end_date': cls.END_DATE,
+             'exchange': 'TEST',
+             'asset_name': 'BENCHMARK'},
+        ], index=cls.sids)
+
     def test_set_symbol_lookup_date(self):
         """
         Test the set_symbol_lookup_date API method.
         """
-        # Note we start sid enumeration at i+3 so as not to
-        # collide with sids [1, 2] added in the setUp() method.
-        dates = pd.date_range('2013-01-01', freq='2D', periods=2, tz='UTC')
-        # Create two assets with the same symbol but different
-        # non-overlapping date ranges.
-        metadata = pd.DataFrame.from_records(
-            [
-                {
-                    'sid': i + 3,
-                    'symbol': 'DUP',
-                    'start_date': date.value,
-                    'end_date': (date + timedelta(days=1)).value,
-                    'exchange': 'TEST',
-                }
-                for i, date in enumerate(dates)
-            ]
-        )
-        with tmp_trading_env(equities=metadata,
-                             load=self.make_load_function()) as env:
-            algo = TradingAlgorithm(env=env)
-
-            # Set the period end to a date after the period end
-            # dates for our assets.
-            algo.sim_params = algo.sim_params.create_new(
-                algo.sim_params.start_session,
-                pd.Timestamp('2015-01-01', tz='UTC')
-            )
+        set_symbol_lookup_date = zipline.api.set_symbol_lookup_date
 
-            # With no symbol lookup date set, we will use the period end date
-            # for the as_of_date, resulting here in the asset with the earlier
-            # start date being returned.
-            result = algo.symbol('DUP')
-            self.assertEqual(result.symbol, 'DUP')
+        def initialize(context):
+            set_symbol_lookup_date(self.asset_ends[0])
+            self.assertEqual(zipline.api.symbol('DUP').sid, self.sids[0])
 
-            # By first calling set_symbol_lookup_date, the relevant asset
-            # should be returned by lookup_symbol
-            for i, date in enumerate(dates):
-                algo.set_symbol_lookup_date(date)
-                result = algo.symbol('DUP')
-                self.assertEqual(result.symbol, 'DUP')
-                self.assertEqual(result.sid, i + 3)
+            set_symbol_lookup_date(self.asset_ends[1])
+            self.assertEqual(zipline.api.symbol('DUP').sid, self.sids[1])
 
             with self.assertRaises(UnsupportedDatetimeFormat):
-                algo.set_symbol_lookup_date('foobar')
+                set_symbol_lookup_date('foobar')
+
+        self.run_algorithm(initialize=initialize)
 
 
-class TestTransformAlgorithm(WithLogger,
```

Comment:
`TestTransformAlgorithm` was kind of a mess. Several of the tests in this suite turned out to be complicated no-ops (e.g., the tests for ordering futures ran an algorithm that only ordered equities).  Also, none of these tests were about the old `transform` API, which is what the name of this suite originally referenced.  It ultimately seemed easier and clearer to just break this up and re-write the tests worth keeping.  Most of the re-written tests now live in `test_ordering.py`.